rootfs/etc/nginx/lua/balancer/static_balancers.lua

```diff
@@ -0,0 +1,78 @@
+local split = require('util.split')
+local ngx_upstream = require("ngx.upstream")
+local implementations = require('balancer.implementations')
+
+local backends = {}
+local balancers = {}
+
+local DEFAULT_LB_ALG = "ewma"
+
+local _M = {}
+
+local function parse_endpoint(endpoint)
+    if (not endpoint.address) or (not endpoint.port) then
```

Comment:
I don't understand this, what is it for? My understanding of the function is it receives a `server` and generates `endpoint` from it and returns that endpoint.

Reply:
Are you asking about the condition on line 13? 

I think that's a fair comment. Seems like I wasn't trusting the abstraction https://github.com/openresty/lua-upstream-nginx-module#get_servers